depends/README.md

```diff
@@ -12,17 +12,13 @@ For example:
 
     make HOST=x86_64-w64-mingw32 -j4
 
-**Bitcoin Core's `configure` script by default will ignore the depends output.** In
+**When configuring Bitcoin Core, CMake by default will ignore the depends output.** In
 order for it to pick up libraries, tools, and settings from the depends build,
-you must set the `CONFIG_SITE` environment variable to point to a `config.site` settings file.
-Make sure that `CONFIG_SITE` is an absolute path.
-In the above example, a file named `depends/x86_64-w64-mingw32/share/config.site` will be
-created. To use it during compilation:
+you must specify the toolchain file.
+In the above example, a file named `depends/x86_64-w64-mingw32/toolchain.cmake` will be
+created. To use it during configuring Bitcoin Core:
 
-    CONFIG_SITE=$PWD/depends/x86_64-w64-mingw32/share/config.site ./configure
-
-The default install prefix when using `config.site` is `--prefix=depends/<host-platform-triplet>`,
-so depends build outputs will be installed in that location.
+    cmake -B build --toolchain depends/x86_64-w64-mingw32/toolchain.cmake
```

Comment:
I noticed that there are still a few leftover `make` reference in this file, and in other documentation: are we planning on updating them here or in a separate pr, e.g. when removing Autotools?

I searched for other possible `make` references revealed a few:
* https://github.com/bitcoin/bitcoin/blob/5ebb4063571aabd89c2d7ed6c2e70d27636efdeb/src/test/README.md?plain=1#L18-L24
* https://github.com/bitcoin/bitcoin/blob/183e2fd6b55d84829faf70ba74d3f8865807772a/src/secp256k1/doc/release-process.md?plain=1#L20-L23
* https://github.com/bitcoin/bitcoin/blob/14093d5d243f6eb9cfef721c80f92848d95032ee/doc/productivity.md?plain=1#L68
* https://github.com/bitcoin/bitcoin/blob/463356d8b8948df118eda269827ca1a6da7d3132/depends/README.md?plain=1#L99
* https://github.com/bitcoin/bitcoin/blob/d8513fe41102dcbfc05235f3b95e33eb1878f880/doc/benchmarking.md?plain=1#L16
* https://github.com/bitcoin/bitcoin/blob/fad444f6e1b1ce5c1572c773db4a9a098c7a0b96/contrib/guix/INSTALL.md?plain=1#L244-L261
* https://github.com/bitcoin/bitcoin/blob/fad444f6e1b1ce5c1572c773db4a9a098c7a0b96/contrib/guix/README.md?plain=1#L263
* https://github.com/bitcoin/bitcoin/blob/fadb06c361bbda78192d89fd06195f0aa5418c56/test/README.md?plain=1#L16
* https://github.com/bitcoin/bitcoin/blob/faeb5b59a098578b3e8c552d35b5ba02b12af14d/README.md?plain=1#L51

Reply:
How much rewrite do we need to do here?
I have pushed https://github.com/hebasto/bitcoin/pull/322 to synchronize on the style, let me know how you imagine it.

Reply:
>    * https://github.com/bitcoin/bitcoin/blob/14093d5d243f6eb9cfef721c80f92848d95032ee/doc/productivity.md?plain=1#L68

See https://github.com/hebasto/bitcoin/pull/323.

Reply:
Resolved in https://github.com/bitcoin/bitcoin/pull/30454#issuecomment-2294786256.